nomad/structs/structs.go

```diff
@@ -4420,7 +4420,11 @@ type Job struct {
 	// Datacenters contains all the datacenters this job is allowed to span
 	Datacenters []string
 
-	// NodePool specifies the node pool this job is allowed to run on
+	// NodePool specifies the node pool this job is allowed to run on.
+	//
+	// An empty value is allowed during job registration, in which case the
+	// namespace default node pool is used in Enterprise and the 'default' node
+	// pool in OSS, but a node pool must be set before the job is stored.
```

Comment:
Just to help out future readers:

```suggestion
	// pool in OSS, but a node pool must be set before the job is stored, so
	// that will happen in admission mutators.
```